L_functions.py

```diff
@@ -0,0 +1,67 @@
+import numpy as np
+import xarray as xr
+from datetime import datetime 
+from datetime import timedelta
+from datetime import date
+import time
+
+def sel_train_data_lead(nc_in_files,var_name,dim_to_stack,target_len,
+                        s_target_date,e_target_date,
+                        rw_1,lead_time,rw,ntimestep):
+    '''
+    This function inputs multiple nc files with the same dimension 
+    (e.g., nc_in_files=path+'PC_serie*.nc') and reads them as a xarray. From these files
+    a predictor array will be created. The dimension to stack the files must be given(e.g.,
+    dim_to_stack='mode') and the variable name of the files (e.g., var_name='pcs').
+    The length of the target must be given (target_len).
+    The start date and end date that we want to predict must be given 
+    (e.g., s_target_date='16-10-1980', e_target_date='16-12-2021') and
+    the running window that was already applied on the predictors with center=False must be
+    declared (rw_1). 
+    
+    The predictor is selected in a way so that the 
+    needed date is predicted at a certain lead time (lead_time) and for a specific running
+    window that was applied on the target (rw). Moreover, a selected time step for the LSTM 
+    is considered (ntimestep).
+    '''
+
+    SDD = int(s_target_date[0:2])
+    SMM = int(s_target_date[3:5])
+    SYY=int(s_target_date[6:10])
+    print('start target date',SDD,SMM,SYY)
+
+    EDD = int(e_target_date[0:2])
+    EMM = int(e_target_date[3:5])
+    EYY = int(e_target_date[6:10])
+    print('end target',EDD,EMM,EYY)
+
+    half_rw = int(rw/2)
+
+    # Import predictors if the predictors have been saved before
+    pc_xr = xr.open_mfdataset(root_results+nc_in_files,
+                              concat_dim=dim_to_stack,combine="nested")# 
```

Comment:
`root_results` will not be found. maybe you provide `pc_xr` as an input to `sel_train_data_lead `